htdocs/core/class/html.formfile.class.php

```diff
@@ -1298,7 +1298,15 @@ public function list_of_documents($filearray, $object, $modulepart, $param = '',
 					if (array_key_exists('rowid', $filearray[$key]) && $filearray[$key]['rowid'] > 0) {
 						$lastrowid = $filearray[$key]['rowid'];
 					}
-					$filepath = $relativepath.$file['name'];
+					//$filepath = $file['level1name'].'/'.$file['name'];
+
+					$filepath = preg_replace('/^'.preg_quote(DOL_DATA_ROOT, '/').'/', '', $file['fullname']);
```

Comment:
Why using the fullname. We should use the short 'name' concatenated to a relativepath (relative path must be a variable set by the server and must not depends on a data coming from the end user as a parameter).
So we must keep use of $relativepath and never $file['fullname']